deploy/upgrade_db_server.yml

```diff
@@ -101,24 +92,17 @@
 
     # Only needed for local testing
     - name: Add citus as a shared preload library
+      become: yes
+      become_method: sudo
       lineinfile:
         dest: "{{ pp_datadir }}/postgresql.conf"
         regexp: "^shared_preload_libraries*"
         line: "shared_preload_libraries = 'citus'"
       when: 'not custom_repo and "citus" in group_names'
 
     - name: Run pg_upgrade consistency check
-      become: yes
-      become_user: "{{ pp_serviceaccount }}"
       command: >
-               {{ pp_prefix }}/{{ pp_install_path }}/bin/pg_upgrade
-                  --old-datadir {{ pp_old_datadir }}
-                  --new-datadir {{ pp_datadir }}
-                  --old-bindir {{ pp_prefix }}/{{ pp_old_install_path }}/bin
-                  --new-bindir {{ pp_prefix }}/{{ pp_install_path }}/bin
-                  --link
-                  -U {{ pp_serviceaccount }}
-                  --check
+          sudo su - {{ pp_serviceaccount }} -c "{{ pp_prefix }}/{{ pp_install_path }}/bin/pg_upgrade --old-datadir {{ pp_old_datadir }} --new-datadir {{ pp_datadir }} --old-bindir {{ pp_prefix }}/{{ pp_old_install_path }}/bin --new-bindir {{ pp_prefix }}/{{ pp_install_path }}/bin --link -U {{ pp_serviceaccount }} --check"
```

Comment:
Minor quibble, but would you mind keeping the newlines?  It makes this much easier to read.